muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/dialog.tsx

```diff
@@ -136,71 +138,77 @@ export default class Dialog extends React.Component<DialogProps, DialogState> {
               ? [this.props.modifier]
               : this.props.modifier;
           return (
-            <div
-              className={`dialog ${(modifiers || [])
-                .map((s) => `dialog--${s}`)
-                .join(" ")} ${this.state.visible ? "dialog--visible" : ""}`}
-              onClick={
-                closeOnOverlayClick
-                  ? this.onOverlayClick.bind(this, closePortal)
-                  : null
-              }
-            >
-              {/* Execution container is missing from here */}
-              <section
+            <FocusTrap active={this.state.visible}>
```

Comment:
FocusTrap prevents second dialog from opening which is used when creating a new purchase for study time period increase.

Problem might be somewhat related to UX/UI design also as I think opening dialog within a dialog is a bit questionable and always harder to handle accessibility wised. Currently there's nothing to be done about it and FocusTrap needs to be reworked to support this functionality but in the future I think we should avoid this kind of thing.